diff --git a/src/backends/meta-color-device.c b/src/backends/meta-color-device.c
index 49f70dcf2..317a1e1e7 100644
--- a/src/backends/meta-color-device.c
+++ b/src/backends/meta-color-device.c
@@ -105,10 +105,12 @@ generate_cd_device_id (MetaMonitor *monitor)
   const char *vendor;
   const char *product;
   const char *serial;
+  const char *connector;
 
   vendor = meta_monitor_get_vendor (monitor);
   product = meta_monitor_get_product (monitor);
   serial = meta_monitor_get_serial (monitor);
+  connector = meta_monitor_get_connector (monitor);
 
   device_id = g_string_new ("xrandr");
 
@@ -135,6 +137,8 @@ generate_cd_device_id (MetaMonitor *monitor)
   if (serial)
     g_string_append_printf (device_id, "-%s", serial);
 
+  g_string_append_printf (device_id, "-%s", connector);
+
 out:
   return g_string_free (device_id, FALSE);
 }
diff --git a/src/backends/meta-color-manager.c b/src/backends/meta-color-manager.c
index 7a583789b..c28315afd 100644
--- a/src/backends/meta-color-manager.c
+++ b/src/backends/meta-color-manager.c
@@ -142,11 +142,13 @@ generate_monitor_id (MetaMonitor *monitor)
   const char *vendor;
   const char *product;
   const char *serial;
+  const char *connector;
   GString *id;
 
   vendor = meta_monitor_get_vendor (monitor);
   product = meta_monitor_get_product (monitor);
   serial = meta_monitor_get_serial (monitor);
+  connector = meta_monitor_get_connector (monitor);
   if (!vendor && !product && !serial)
     return g_strdup (meta_monitor_get_connector (monitor));
 
@@ -158,6 +160,8 @@ generate_monitor_id (MetaMonitor *monitor)
     g_string_append_printf (id, "%sp:%s", id->len > 0 ? ";" : "", product);
   if (serial)
     g_string_append_printf (id, "%sp:%s", id->len > 0 ? ";" : "", serial);
+  if (connector)
+    g_string_append_printf (id, "%sc:%s", id->len > 0 ? ";" : "", connector);
 
   return g_string_free (id, FALSE);
 }
